packages/core/src/api/toDecimal.ts

```diff
@@ -32,9 +41,15 @@ export function toDecimal<TAmount, TOutput>(calculator: Calculator<TAmount>) {
     assert(isDecimal, NON_DECIMAL_CURRENCY_MESSAGE);
 
     const units = toUnitsFn(dineroObject);
+    const isNegative = units.some((unit) => lessThanFn(signFn(unit), zero));
 
     const getDecimalFn = getDecimal(calculator, dineroObject.formatter);
-    const value = getDecimalFn(units, scale);
+    const decimalValue = getDecimalFn(units, scale);
+
+    // A leading negative zero is a special case because the toString
+    // formatter (generally String) won't preserve its negative sign.
+    const value =
+      isNegative && equalFn(units[0], zero) ? `-${decimalValue}` : decimalValue;
```

Comment:
I could also use `isZero` here, but we already have all of the components at hand to perform this check. Let me know if `isZero` is preferred.

Reply:
Importing `isZero` might make `toDecimal` slightly heavier, so I prefer your approach.